tools/wasi/uvwasi.cpp

```diff
@@ -0,0 +1,81 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2021 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "uvwasi.hpp"
+#include <uvwasi.h>
+
+namespace fizzy::wasi
+{
+namespace
+{
+class UVWASIImpl final : public UVWASI
+{
+    /// UVWASI state.
+    uvwasi_t m_state{};
+    bool m_inited = false;
+
+public:
+    ~UVWASIImpl() final
+    {
+        if (m_inited)
```

Comment:
Also suggested a change to uvwasi to handle this https://github.com/nodejs/uvwasi/pull/163

But it won't help the case with calling `init` twice, so we need a flag anyway.